app/helpers/thredded/urls_helper.rb

```diff
@@ -97,7 +97,9 @@ def edit_preferences_path(messageboard = nil, params = {})
     # @param messageboard [Thredded::Messageboard, nil]
     # @return [String] the path to the global or messageboard search.
     def search_path(messageboard = nil)
```

Comment:
I think it would be better to render the search form from every view that needs it explicitly, and have separate partials for topics and private topics searches. We'll also need different text for the private topics search form, something like  "Search Messages", so it makes sense to separate them.